### PR TITLE
[ACTION] Webflow new actions #1488

### PR DESCRIPTION
**Pull request by lcaresia**

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

File: components/webflow/webflow.app.mjs [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+        $,
+        config: {
+          params: {
+            status: status,
+            offset: page ?? 0,
+            limit: 100,
+          },
+        },
+      });
+    },
```
Comment (jcortes) [inline components/webflow/webflow.app.mjs:149]: If you remove the `axios` lib there is a way to use the `get` method using the sdk client lib in order to replace these functions

File: components/webflow/webflow.app.mjs [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+    }) {
+      return await axios($ ?? this, {
+        url: `${this._apiUrl()}${path}`,
+        headers: {
+          "Authorization": `Bearer ${this._authToken()}`,
+          "accept-version": "1.0.0",
+        },
+        ...config,
+      });
+    },
```
Comment (jcortes) [inline components/webflow/webflow.app.mjs:101]: If you decide to remove `axios` import you can also remove these functions

File: components/webflow/actions/create-collection-item/create-collection-item.mjs [hunk 3]
```diff
@@ -1,53 +1,59 @@
-// legacy_hash_id: a_q1ioGv
-import { axios } from "@pipedream/platform";
+import webflow from "../../webflow.app.mjs";
 
 export default {
   key: "webflow-create-collection-item",
-  name: "Create Item",
+  name: "Create Collection Item",
   description: "Create new collection item",
```
Comment (jcortes) [inline components/webflow/actions/create-collection-item/create-collection-item.mjs:6]: I think you can add the docs like
```suggestion
  description: "Create new collection item. [See the docs here](https://developers.webflow.com/#create-new-collection-item)",
```
and the same goes for the rest of components that don't have any doc reference in the `description` property

File: components/webflow/actions/create-collection-item/create-collection-item.mjs [hunk 4]
```diff
@@ -1,53 +1,59 @@
...
-  version: "0.1.1",
+  version: "0.2.1",
```
Comment (jcortes) [inline components/webflow/actions/create-collection-item/create-collection-item.mjs:7]: I don't really see the reason on changing the `MINOR` version to `2` I would rather change the `PATCH` version right? like
```suggestion
  version: "0.1.2",
```
the same goes to the rest of components that were modified

File: components/webflow/actions/create-collection-item/create-collection-item.mjs [hunk 5]
```diff
@@ -1,53 +1,59 @@
...
     },
-    _archived: {
-      type: "boolean",
-    },
-    _draft: {
-      type: "boolean",
-    },
   },
-  async run({ $ }) {
+  async run() {
```
Comment (jcortes) [inline components/webflow/actions/create-collection-item/create-collection-item.mjs:44]: We try to always add the `$` object in order to export a summary string to let the user know about the response like in this example
```javascript
  async run({ $ }) {
    const webflow = this.webflow._createApiClient();

    const response = await webflow.createItem({
      collectionId: this.collectionId,
      fields: {
        name: this.name,
        slug: this.slug,
        _archived: false,
        _draft: false,
      },
    }, {
      live: this.live,
    });

    $.export("$summary", `Successfully created item ${response._id}.`);

    return response;
  },
```
the same goes to the rest of components in this app

File: components/webflow/actions/fulfill-order/fulfill-order.mjs [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    },
+    sendOrderFulfilledEmail: {
+      label: "Send Order Fulfilled Email",
+      description: "Whether or not the Order Fulfilled email should be sent",
+      type: "boolean",
+      default: false,
+    },
+  },
+  async run({ $ }) {
+    return this.webflow._makeRequest(`/sites/${this.siteId}/order/${this.orderId}/fulfill`, {
```
Comment (jcortes) [inline components/webflow/actions/fulfill-order/fulfill-order.mjs:31]: Not sure but I guess you can replace this by calling the sdk client lib with `post` method. The same goes for the rest of components calling `axios` lib through `this._makeRequest` function

File: components/webflow/actions/list-orders/list-orders.mjs [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+      description: "The status to filter the orders.",
+      type: "string",
+      options: [
+        "pending",
+        "refunded",
+        "dispute-lost",
+        "fulfilled",
+        "disputed",
+        "unfulfilled",
+      ],
```
Comment (jcortes) [inline components/webflow/actions/list-orders/list-orders.mjs:28]: You can set a constants variable for these kind of options in properties. The suggestion is to create a file named `webflow/common/constants.mjs` and there export all your constants like
```suggestion
      options: constants.STATUSES,
```

**Review by jcortes — CHANGES_REQUESTED** [review]
Hi @lcaresia, I've just added a few requested changes. Also can you bring back `package.json` with these contents?
```json
{
    "name": "@pipedream/webflow",
    "version": "0.3.3",
    "description": "Pipedream Webflow Components",
    "main": "webflow.app.mjs",
    "keywords": [
        "pipedream",
        "webflow"
    ],
    "homepage": "https://pipedream.com/apps/webflow",
    "author": "Pipedream <support@pipedream.com> (https://pipedream.com/)",
    "license": "MIT",
    "dependencies": {
        "webflow-api": "^0.5.0"
    },
    "gitHead": "e12480b94cc03bed4808ebc6b13e7fdb3a1ba535",
    "publishConfig": {
        "access": "public"
    }
}
```

File: components/webflow/actions/create-collection-item/create-collection-item.mjs (hunk 5 above)
Comment (lcaresia) [reply components/webflow/actions/create-collection-item/create-collection-item.mjs:44]: Is returning the API Client method directly now, it's ok?

File: components/webflow/actions/create-collection-item/create-collection-item.mjs [hunk 8]
```diff
@@ -1,53 +1,59 @@
...
```
Comment (michelle0927) [inline components/webflow/actions/create-collection-item/create-collection-item.mjs:44]: All actions need to import the `$` variable and export a summary.
```suggestion
  async run({ $ }) {
```

File: components/webflow/actions/create-collection-item/create-collection-item.mjs [hunk 9]
```diff
@@ -1,53 +1,59 @@
...
+      collectionId: this.collectionId,
+      fields: {
+        name: this.name,
+        slug: this.slug,
+        _archived: false,
+        _draft: false,
       },
+    }, {
+      live: this.live,
     });
```
Comment (michelle0927) [inline components/webflow/actions/create-collection-item/create-collection-item.mjs:57]: We should export a summary before returning the results. This applies to all action components.
```suggestion
    const resp = await webflow.createItem({
      collectionId: this.collectionId,
      fields: {
        name: this.name,
        slug: this.slug,
        _archived: false,
        _draft: false,
      },
    }, {
      live: this.live,
    });
    $.export("$summary", `Successfully created item ${this.name}`);
    return resp;
```

File: components/webflow/actions/fulfill-order/fulfill-order.mjs [hunk 10]
```diff
@@ -0,0 +1,39 @@
+import webflow from "../../webflow.app.mjs";
+
+export default {
+  key: "webflow-fulfill-order",
+  name: "Fulfill Order",
+  description: "Fulfill a order. [See the docs here](https://developers.webflow.com/#fulfill-order)",
```
Comment (michelle0927) [inline components/webflow/actions/fulfill-order/fulfill-order.mjs:6]:
```suggestion
  description: "Fulfill an order. [See the docs here](https://developers.webflow.com/#fulfill-order)",
```

File: components/webflow/actions/get-item-inventory/get-item-inventory.mjs [hunk 11]
```diff
@@ -0,0 +1,42 @@
+import webflow from "../../webflow.app.mjs";
+
+export default {
+  key: "webflow-get-item-inventory",
+  name: "Get Item Inventory",
+  description: "Get the inventory of a specify item. [See the docs here](https://developers.webflow.com/#item-inventory)",
```
Comment (michelle0927) [inline components/webflow/actions/get-item-inventory/get-item-inventory.mjs:6]:
```suggestion
  description: "Get the inventory of a specific item. [See the docs here](https://developers.webflow.com/#item-inventory)",
```

File: components/webflow/actions/get-order/get-order.mjs [hunk 12]
```diff
@@ -0,0 +1,30 @@
+import webflow from "../../webflow.app.mjs";
+
+export default {
+  key: "webflow-get-order",
+  name: "Get Order",
+  description: "Get a order. [See the docs here](https://developers.webflow.com/#get-order)",
```
Comment (michelle0927) [inline components/webflow/actions/get-order/get-order.mjs:6]:
```suggestion
  description: "Get an order. [See the docs here](https://developers.webflow.com/#get-order)",
```

File: components/webflow/actions/refund-order/refund-order.mjs [hunk 13]
```diff
@@ -0,0 +1,29 @@
+import webflow from "../../webflow.app.mjs";
+
+export default {
+  key: "webflow-refund-order",
+  name: "Refund Order",
+  description: "Refund a order. [See the docs here](https://developers.webflow.com/#refund-order)",
```
Comment (michelle0927) [inline components/webflow/actions/refund-order/refund-order.mjs:6]:
```suggestion
  description: "Refund an order. [See the docs here](https://developers.webflow.com/#refund-order)",
```

File: components/webflow/actions/unfulfill-order/unfulfill-order.mjs [hunk 14]
```diff
@@ -0,0 +1,29 @@
+import webflow from "../../webflow.app.mjs";
+
+export default {
+  key: "webflow-unfulfill-order",
+  name: "Unfulfill Order",
+  description: "Unfulfill a order. [See the docs here](https://developers.webflow.com/#unfulfill-order)",
```
Comment (michelle0927) [inline components/webflow/actions/unfulfill-order/unfulfill-order.mjs:6]:
```suggestion
  description: "Unfulfill an order. [See the docs here](https://developers.webflow.com/#unfulfill-order)",
```

File: components/webflow/actions/update-item-inventory/update-item-inventory.mjs [hunk 15]
```diff
@@ -0,0 +1,77 @@
+import webflow from "../../webflow.app.mjs";
+
+export default {
+  key: "webflow-update-item-inventory",
+  name: "Update Item Inventory",
+  description: "Update the inventory of a specify item. [See the docs here](https://developers.webflow.com/#update-item-inventory)",
```
Comment (michelle0927) [inline components/webflow/actions/update-item-inventory/update-item-inventory.mjs:6]:
```suggestion
  description: "Update the inventory of a specific item. [See the docs here](https://developers.webflow.com/#update-item-inventory)",
```

File: components/webflow/actions/update-order/update-order.mjs [hunk 16]
```diff
@@ -0,0 +1,61 @@
+import webflow from "../../webflow.app.mjs";
+
+export default {
+  key: "webflow-update-order",
+  name: "Update Order",
+  description: "Update a order. [See the docs here](https://developers.webflow.com/#update-order)",
```
Comment (michelle0927) [inline components/webflow/actions/update-order/update-order.mjs:6]:
```suggestion
  description: "Update an order. [See the docs here](https://developers.webflow.com/#update-order)",
```

**Review by michelle0927 — CHANGES_REQUESTED** [review]

**Review by michelle0927 — APPROVED** [review]

**Pull request merged**